src/taipy/core/_repository/_filesystem_repository.py

```diff
@@ -64,7 +66,7 @@ def _save(self, entity: Entity):
     def _exists(self, entity_id: str) -> bool:
         return self.__get_path(entity_id).exists()
 
-    @_retry_read_entity((Exception,))
+    @_retry_read_entity(__EXCEPTIONS_TO_RETRY)
     def _load(self, entity_id: str) -> Entity:
         try:
             with pathlib.Path(self.__get_path(entity_id)).open(encoding="UTF-8") as source:
```

Comment:
```suggestion
    path = pathlib.Path(self.__get_path(entity_id))
    if path.is_file():
        try:
            with pathlib.Path(self.__get_path(entity_id)).open(encoding="UTF-8") as source:
```

Reply:
The file-not-exists case is already caught by the `except FileNotFound:` right?

Reply:
short answer is Yes 😃 long answer is, you will still need to try and "read" it to know it doesn't exist. Maybe we should introduce a new Exception here "FileCannotBeRead" for example. 
- If path == False: Raise "FileNotFoundError" -> don't retry (no point trying to read a non-existing file again and again anyway) 
- If path == True: Try read the file, if cant -> raise"FileCannotBeRead" and retry

What do you think? @trgiangdo @jrobinAV

Reply:
Interesting idea. That would help us having more control on when to retry.

Reply:
I believe it s a great idea !